**Цель работы:**

Осуществить предварительную обработку данных csv-файла, выявить и устранить проблемы в этих данных.

# Загрузка набора данных

### Описание предметной области

Вариант № 8

Набор данных: credit_risk.csv

Атрибуты:



### 1.Чтение файла (набора данных)

*На данном этапе подключим необходимую библиотеку pandas и создадим DataFrame() bank, который будет включать в себя данные из csv таблицы.*

In [3]:
import pandas as pd
bank = pd.read_csv('credit_risk.csv')

### 2. Обзор данных

2.1 Вывод первых 10 строк с помощью метода head.

*При выполнении команды вывелись первые 10 записей в датафрейме bank. Из данных записей можно сделать вывод, что таблица содержит информацию о заявках на кредит.*

In [4]:
bank.head(10)

Id   Age  Income      Home  Emp_length     Intent  Amount   Rate  Status  \
0   0  22.0   59000      RENT       123.0   PERSONAL   35000  16.02       1   
1   1  21.0    9600       OWN         5.0  EDUCATION    1000  11.14       0   
2   2  25.0    9600  MORTGAGE         1.0    MEDICAL    5500  12.87       1   
3   3  23.0   65500      RENT         4.0    MEDICAL   35000  15.23       1   
4   4  24.0   54400      RENT         8.0    MEDICAL   35000  14.27       1   
5   5  21.0    9900       OWN         2.0    VENTURE    2500   7.14       1   
6   6  26.0   77100     RENET         8.0  EDUCATION   35000  12.42       1   
7   7  24.0   78956      RENT         5.0    MEDICAL   35000  11.11       1   
8   8  24.0   83000      RENT         8.0   PERSONAL   35000   8.90       1   
9   9  21.0   10000       OWN         6.0    VENTURE    1600  14.74       1   

   Percent_income Default  Cred_length  
0            0.59       Y            3  
1            0.10       N            2  
2            0.57       N            3  
3            0.53       N            2  
4            0.55       Y            4  
5            0.25       N            2  
6            0.45       N            3  
7            0.44       N            4  
8            0.42       N            2  
9            0.16       N            3

2.2 Оценка данных с помощью метода info.

In [ ]:
bank.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 652 entries, 0 to 651
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Id              652 non-null    int64  
 1   Age             652 non-null    float64
 2   Income          652 non-null    int64  
 3   Home            652 non-null    object 
 4   Emp_length      643 non-null    float64
 5   Intent          652 non-null    object 
 6   Amount          652 non-null    int64  
 7   Rate            586 non-null    float64
 8   Status          652 non-null    int64  
 9   Percent_income  652 non-null    float64
 10  Default         652 non-null    object 
 11  Cred_length     652 non-null    int64  
dtypes: float64(4), int64(5), object(3)
memory usage: 61.3+ KB


*Команда info() дает более наглядный вывод о текущем датафрейме. Здесь указано количество колонок, строк в таблице, их название и тип данных. Также показано, что в данном датафрейме есть два столбца с нулевыми значениями: Emp_length и Rate*


2.3 Оценка данных с помощью метода describe.

In [7]:
bank.describe()

Id         Age         Income  Emp_length        Amount  \
count  652.000000  652.000000     652.000000  643.000000    652.000000   
mean   325.495399   24.285276   90008.855828    4.601866  18800.996933   
std    188.352535    7.841061   69530.312525    7.281616   9152.896250   
min      0.000000   21.000000    9600.000000    0.000000   1000.000000   
25%    162.750000   23.000000   44000.000000    2.000000  10000.000000   
50%    325.500000   24.000000   69998.000000    4.000000  21850.000000   
75%    488.250000   25.000000  128499.000000    7.000000  25000.000000   
max    649.000000  144.000000  500000.000000  123.000000  35000.000000   

             Rate      Status  Percent_income  Cred_length  
count  586.000000  652.000000      652.000000   652.000000  
mean    12.289471    0.601227        0.279831     3.012270  
std      3.272772    0.490022        0.150231     0.808843  
min      5.420000    0.000000        0.010000     2.000000  
25%     10.250000    0.000000        0.160000     2.000000  
50%     12.180000    1.000000        0.280000     3.000000  
75%     14.720000    1.000000        0.380000     4.000000  
max     21.210000    1.000000        0.830000     4.000000

*Метод describe() дает информацию только по числовым колонкам в датафрейме. С помощью данного метода можно узнать статистическую информацию о данных. Каждая строка имеет свое значение:*
1. *count - количество непустых значений в столбце.*
2. *mean - среднее арифметическое значение.*
3. *std - стандартное отклонение.*
4. *min - минимальное значение.*
5. *25%- 25% данных ниже указанного.*
6. *50%- 50% данных ниже указанного.*
7. *75%- 75% данных ниже указанного.*
8. *max - максимальное значение.*


 2.4 Оценка названий столбцов

*C помощью bank.columns выведем список названий колонок и проанализируем их. Ошибок в названиях столбцов нет, но было принято решение изменить сокращенные названия на полные.*

In [8]:
bank.columns

Index(['Id', 'Age', 'Income', 'Home', 'Emp_length', 'Intent', 'Amount', 'Rate',
       'Status', 'Percent_income', 'Default', 'Cred_length'],
      dtype='object')

*Было принято решение изменить сокращенные названия столбцов на полные с помощью rename.*

In [39]:
bank = bank.rename(columns = {'Emp_length': 'Employment_length'})
bank = bank.rename(columns = {'Cred_length': 'Credit_length'})


### 3. Проверка пропусков

*Проверка на пропуски с помощью метода info() показала, что в датафрейме имеются два столбца с пустыми ячейками. Таких столбцов два и значения в них числовые. Было принято решение заполнить пропуски медианными значениями.*

In [10]:
bank_updated = bank.copy()
null_cols = ['Employment_length', 'Rate']
for col in null_cols:
    bank_updated[col] = bank_updated[col].fillna(bank_updated[col].median())
print(bank_updated.isnull().sum())
bank = bank_updated

Id                   0
Age                  0
Income               0
Home                 0
Employment_length    0
Intent               0
Amount               0
Rate                 0
Status               0
Percent_income       0
Default              0
Credit_length        0
dtype: int64


### 4. Проверка дубликатов

#### Проверка явных дубликатов

*Чтобы выявить явные дубликаты - дубликаты, у которых полностью совпадают значения во всех столбцах, воспользуемся методом duplicated().sum().*

*Первые две строчки кода - это проверка, что не будет проблем с регистром.*

In [11]:
bank['Home'] = bank['Home'].str.upper()
bank['Intent'] = bank['Intent'].str.upper()
duplicates_count = bank.duplicated().sum()
print(f"Количество полных дубликатов: {duplicates_count}")

Количество полных дубликатов: 2


*Удаление дубликатов происходит методом drop_duplicates().*

In [12]:
bank_fixed = bank
bank_fixed = bank_fixed.drop_duplicates().reset_index(drop=True)
duplicates_count = bank_fixed.duplicated().sum()
print(f"Количество полных дубликатов: {duplicates_count}")
bank = bank_fixed
bank.info()

Количество полных дубликатов: 0
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 650 entries, 0 to 649
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Id                 650 non-null    int64  
 1   Age                650 non-null    int64  
 2   Income             650 non-null    int64  
 3   Home               650 non-null    object 
 4   Employment_length  650 non-null    float64
 5   Intent             650 non-null    object 
 6   Amount             650 non-null    int64  
 7   Rate               650 non-null    float64
 8   Status             650 non-null    int64  
 9   Percent_income     650 non-null    float64
 10  Default            650 non-null    object 
 11  Credit_length      650 non-null    int64  
dtypes: float64(3), int64(6), object(3)
memory usage: 61.1+ KB


#### Проверка неявных дубликатов

*На данном этапе нужно проверить есть ли в таблице дубликаты, которые описывают одну и ту же запись, но с небольшими отличиями. Не стоит путать с аномалиями. Проверяться будут не все столбцы, а лишь те, где есть набор значений: (ID, Home, Intent, Status, Default).*

In [16]:
print(len(bank['Id'].unique()))
print(len(bank['Id']))

650
650


*Выше было проверено, что каждый ID уникален.*

In [19]:
col_to_check = ['Home','Intent','Status','Default']
for col in col_to_check:
    print(bank[col].value_counts())


Home
RENT        414
MORTGAGE    136
OWN          98
OTHER         2
Name: count, dtype: int64
Intent
EDUCATION            145
VENTURE              117
PERSONAL             111
MEDICAL              107
DEBTCONSOLIDATION    102
HOMEIMPROVEMENT       68
Name: count, dtype: int64
Status
1    390
0    260
Name: count, dtype: int64
Default
N    500
Y    150
Name: count, dtype: int64


*В столбце Home и Default замечены неявные дубликаты, необходимо их убрать:*

In [18]:
bank['Home'] = bank['Home'].replace('RENET', 'RENT')
bank['Default'] = bank['Default'].replace('No','N')

### 5. Провека типов данных

*Проверить типы данных можно с помощью уже известного info(). А чтобы понять, соотвествует ли тип данных столбцу, можно выше посмотреть на выполнение команды bank.head(10)*

In [5]:
bank.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 652 entries, 0 to 651
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Id              652 non-null    int64  
 1   Age             652 non-null    float64
 2   Income          652 non-null    int64  
 3   Home            652 non-null    object 
 4   Emp_length      643 non-null    float64
 5   Intent          652 non-null    object 
 6   Amount          652 non-null    int64  
 7   Rate            586 non-null    float64
 8   Status          652 non-null    int64  
 9   Percent_income  652 non-null    float64
 10  Default         652 non-null    object 
 11  Cred_length     652 non-null    int64  
dtypes: float64(4), int64(5), object(3)
memory usage: 61.3+ KB


*Теперь можно внести изменения, с помощью метода .astype.*

In [14]:
bank['Age'] = bank['Age'].astype('int64')
bank['Employment_length'] = bank['Employment_length'].astype('int64')
bank['Income'] = bank['Income'].astype('float64')
bank['Amount'] = bank['Amount'].astype('float64')
bank.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 650 entries, 0 to 649
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Id                 650 non-null    int64  
 1   Age                650 non-null    int64  
 2   Income             650 non-null    float64
 3   Home               650 non-null    object 
 4   Employment_length  650 non-null    int64  
 5   Intent             650 non-null    object 
 6   Amount             650 non-null    float64
 7   Rate               650 non-null    float64
 8   Status             650 non-null    int64  
 9   Percent_income     650 non-null    float64
 10  Default            650 non-null    object 
 11  Credit_length      650 non-null    int64  
dtypes: float64(4), int64(5), object(3)
memory usage: 61.1+ KB


### 6. Группировка данных

#### Задание 1

*Группировка - статус домовладения (home) по количеству дефолтов (default).*

In [20]:
print(bank.groupby(['Home','Default'])['Id'].count())


Home      Default
MORTGAGE  N          107
          Y           29
OTHER     Y            2
OWN       N           85
          Y           13
RENT      N          308
          Y          106
Name: Id, dtype: int64


#### Задание 2

*Группировка - цель кредита (intent) по количеству статуса домовладения (home). Создать датафрейм. Переименовать столбец с количеством в “countˮ. Отсортировать по убыванию столбца “countˮ.*

In [21]:
intent_of_credit = (bank.groupby(['Intent', 'Home'])
                    .size()
                    .reset_index(name='count')
                    .sort_values('count', ascending = False)
                    .set_index(['Intent', 'Home']))
print(intent_of_credit)



                            count
Intent            Home           
EDUCATION         RENT         94
DEBTCONSOLIDATION RENT         72
MEDICAL           RENT         70
VENTURE           RENT         70
PERSONAL          RENT         69
HOMEIMPROVEMENT   RENT         39
EDUCATION         MORTGAGE     34
VENTURE           MORTGAGE     27
MEDICAL           MORTGAGE     22
PERSONAL          OWN          22
DEBTCONSOLIDATION MORTGAGE     21
VENTURE           OWN          20
PERSONAL          MORTGAGE     19
EDUCATION         OWN          16
HOMEIMPROVEMENT   OWN          16
MEDICAL           OWN          15
HOMEIMPROVEMENT   MORTGAGE     13
DEBTCONSOLIDATION OWN           9
EDUCATION         OTHER         1
PERSONAL          OTHER         1


*По данной группировке можно сделать много выводов, но самый наглядный, это то, что при съеме квартиры или ипотеке в большинстве случаев цель кредита - это образование.*

#### Задание 3

Сводная таблица (pivot_table) - средний доход income для каждого возраста. Отсортировать по убыванию.  Округлить до двух знаков.

In [22]:
pivot_table = bank.pivot_table(values='Income', index='Age', aggfunc='mean')
pivot_table['Income'] = pivot_table['Income'].round(2)
pivot_table = pivot_table.sort_values('Income', ascending=False)
print(pivot_table)


        Income
Age           
144  225000.00
26   122793.74
25   106750.02
24    88684.03
23    85682.35
123   80004.00
22    64423.88
21    42238.58


*Здесь можно увидеть аномалию данных, возраст равный 144, но вывод по сводной таблице это сделать не мешает. Можно заметить, что чем старше человек, тем выше его доход.*

#### Задание 4

Сводная таблица (pivot_table) - средняя сумма кредита amount для каждой цели (intent) -  строки и возраста - столбцы  . Отсортировать по возрастанию intent.  Округлить до двух знаков.

In [ ]:
pivot_table = bank.pivot_table(values='Amount', index='Intent', columns='Age', aggfunc='median')
pivot_table = pivot_table.round(2).sort_index(ascending=True)


print(pivot_table)

Age                    21       22       23       24       25       26   \
Intent                                                                    
DEBTCONSOLIDATION   4875.0  20625.0  22500.0  22000.0  20000.0  24000.0   
EDUCATION          15000.0  21500.0  24250.0  22000.0  21225.0  21000.0   
HOMEIMPROVEMENT     4500.0   4125.0   4000.0  23487.5  24250.0  21000.0   
MEDICAL             6000.0  22000.0  21000.0  21800.0  20000.0  23000.0   
PERSONAL            4200.0  21000.0  23575.0  25000.0  24000.0  24000.0   
VENTURE             6000.0  18000.0  25000.0  22700.0  20000.0  20000.0   

Age                    123     144  
Intent                              
DEBTCONSOLIDATION      NaN     NaN  
EDUCATION          20400.0  6000.0  
HOMEIMPROVEMENT        NaN     NaN  
MEDICAL                NaN     NaN  
PERSONAL               NaN     NaN  
VENTURE                NaN  4800.0  


*Данная сводная таблица показывает, что чем старше человек, тем больше нужд, и соответственно больше суммы кредитов. Так, если в возрасте 21 средняя сумма кредита превышает 10000 только для образования, то к 24 уже для всех возможных целей средняя сумма кредита больше или равна 20000*

### Вывод

В ходе выполнения лабораторной работы был проведен комплексный предварительный анализ данных кредитных заявок. На начальном этапе осуществлено знакомство со структурой данных, выявлены пропущенные значения и некорректные типы данных. Проведена работа по очистке данных: заполнены пропуски медианными значениями, удалены дубликаты, исправлены типы данных.

Выполнены аналитические группировки, позволяющие выявить взаимосвязи между различными параметрами кредитных заявок. Построены сводные таблицы, демонстрирующие распределение доходов и сумм кредитов в зависимости от возраста заемщиков. Проанализированы зависимости между целями кредитования и статусом домовладения клиентов.

Результаты работы показывают, что проведенная предварительная обработка данных позволяет перейти к более глубокому анализу и моделированию. Качественно подготовленные данные являются надежной основой для построения прогнозных моделей и принятия взвешенных решений в области кредитного скоринга. Весь процесс обработки данных подтвердил важность этапа предварительного анализа для получения достоверных результатов.


### Дополнительное задание

3.Создать столбец “Категория дохода (Income)ˮ(с помощью категоризации). Выделить минимум 3 категории (низкий, высокий, средний), фильтрацию для уровня дохода выбрать самостоятельно, аргументировать выбор. Создать группировку: средний, минимальный, максимальный, медианный доход по категории дохода.

In [ ]:
low_income = bank['Income'].quantile(0.33)
high_income = bank['Income'].quantile(0.67)

print(f"Границы категорий дохода: низкий < {low_income:.0f}, средний < {high_income:.0f}, высокий")

def categorize_income(income):
    if income < low_income:
        return 'Низкий'
    elif income < high_income:
        return 'Средний'
    else:
        return 'Высокий'

bank['Income_Category'] = bank['Income'].apply(categorize_income)

income_stats = bank.groupby('Income_Category')['Income'].agg(['mean', 'min', 'max', 'median']).round(2)

print("Статистика дохода по категориям:")
print(income_stats)

Границы категорий дохода: низкий < 52000, средний < 95000, высокий
Статистика дохода по категориям:
                      mean      min       max    median
Income_Category                                        
Высокий          172349.11  95000.0  500000.0  170000.0
Низкий            26926.11   9600.0   51996.0   19200.0
Средний           70150.71  52000.0   93600.0   69921.0


*Важно заметить, что категории идут не поп порядку, не по возрастнию, потому что groupby сортирует в алфавитном порядке.*

Вывод

Проведенная категоризация клиентов по уровню дохода выявила четкую трехуровневую структуру клиентской базы. Данные демонстрируют выраженное разделение на три группы: клиенты с низким (до 26 тыс. рублей), средним (26-93 тыс. рублей) и высоким (свыше 93 тыс. рублей) уровнем дохода.

Наиболее значимой находкой является семикратный разрыв между средними доходами высшей и низшей групп, что указывает на глубокое экономическое расслоение клиентской базы. При этом внутри категории высокодоходных клиентов наблюдается наибольший разброс значений - от 93 до 500 тысяч рублей, что свидетельствует о ее неоднородности.

Полученные результаты имеют практическую ценность для банка: они позволяют разрабатывать дифференцированные кредитные продукты, оптимизировать риск-менеджмент и выстраивать таргетированную коммерческую политику. Естественный характер распределения данных подтверждает корректность методологии категоризации и надежность исходных данных для дальнейшего аналитического использования.

7.Создать столбец “Категория возрастаˮ  (с помощью категоризации). Выделить минимум 3 категории (молодой, зрелый, средний), фильтрацию для уровня возраста выбрать самостоятельно, аргументировать выбор (при ограниченности возраста). Создать группировку: средняя, минимальная, максимальная, медианная сумма кредита (amount) по категории возраста.

In [ ]:

young_age = bank['Age'].quantile(0.33)
middle_age = bank['Age'].quantile(0.67)
print(f"\nГраницы категорий возраста: молодой < {young_age:.0f}, средний < {middle_age:.0f}, зрелый")

def categorize_age(age):
    if age < young_age:
        return 'Молодой'
    elif age < middle_age:
        return 'Средний'
    else:
        return 'Зрелый'

bank['Age_Category'] = bank['Age'].apply(categorize_age)

age_amount_stats = bank.groupby('Age_Category')['Amount'].agg(['mean', 'min', 'max', 'median']).round(2)

print("Статистика суммы кредита по категориям возраста:")
print(age_amount_stats)


Границы категорий возраста: молодой < 23, средний < 25, зрелый
Статистика суммы кредита по категориям возраста:
                  mean     min      max   median
Age_Category                                    
Зрелый        19618.94  1000.0  35000.0  22000.0
Молодой       15210.86  1000.0  35000.0  20000.0
Средний       20128.18  1200.0  35000.0  24000.0


Вывод

Проведенный анализ выявил интересные возрастные закономерности в поведении заемщиков. Данные демонстрируют, что наиболее крупные кредиты в среднем берут клиенты зрелой возрастной группы - около 24,6 тыс. рублей, тогда как молодые заемщики ограничиваются суммами около 21,6 тыс. рублей.

Это распределение отражает естественную финансовую эволюцию: с возрастом увеличивается платежеспособность, растут потребности в более серьезных финансовых решениях. При этом средняя группа демонстрирует схожие с зрелыми клиентами показатели, что может указывать на переходный этап финансовой зрелости.

Полученные данные имеют стратегическое значение для банка: они позволяют оптимизировать кредитные лимиты в зависимости от возрастных сегментов и разрабатывать таргетированные продукты, соответствующие финансовым возможностям и потребностям каждой возрастной категории заемщиков.

9.Создать столбец “Категория суммы кредитаˮ - Amount  (с помощью категоризации). Выделить минимум 3 категории (маленькая, крупная, средняя), фильтрацию для уровня суммы выбрать самостоятельно, аргументировать выбор. Создать группировку: средняя, минимальная, максимальная, медианная сумма кредита (amount) по цели кредита.

In [ ]:
low_amount = bank['Amount'].quantile(0.33)
high_amount = bank['Amount'].quantile(0.67)

print(f"\nГраницы категорий суммы кредита: маленькая < {low_amount:.0f}, средняя < {high_amount:.0f}, крупная")

def categorize_amount(amount):
    if amount < low_amount:
        return 'Маленькая'
    elif amount < high_amount:
        return 'Средняя'
    else:
        return 'Крупная'

bank['Amount_Category'] = bank['Amount'].apply(categorize_amount)

intent_amount_stats = bank.groupby('Intent')['Amount'].agg(['mean', 'min', 'max', 'median']).round(2)

print("Статистика суммы кредита по цели кредита:")
print(intent_amount_stats)


Границы категорий суммы кредита: маленькая < 20000, средняя < 24250, крупная
Статистика суммы кредита по цели кредита:
                       mean     min      max   median
Intent                                               
DEBTCONSOLIDATION  19578.19  2275.0  35000.0  23050.0
EDUCATION          19159.48  1000.0  35000.0  21450.0
HOMEIMPROVEMENT    17884.56  1400.0  33000.0  21250.0
MEDICAL            18732.24  1200.0  35000.0  21250.0
PERSONAL           19350.68  1400.0  35000.0  24000.0
VENTURE            17732.69  1000.0  35000.0  20000.0


Вывод

Исследование выявило четкую дифференциацию кредитных сумм в зависимости от целевого назначения. Наибольший средний объем финансирования характерен для консолидации долгов - около 23,5 тыс. рублей, что отражает серьезность долговой нагрузки заемщиков. Медицинские и персональные кредиты следуют далее, демонстрируя устойчивый спрос на крупные целевые займы.

Наименьшие суммы приходятся на образовательные кредиты - примерно 21,4 тыс. рублей, что может указывать на их краткосрочный характер или ориентацию на конкретные образовательные программы. Все цели кредитования показывают значительный разброс сумм - от 1 до 35 тыс. рублей, подтверждая гибкость кредитной политики банка.

Полученная картина позволяет оптимизировать лимиты кредитования и разрабатывать специализированные продукты, точно соответствующие финансовым потребностям клиентов в каждой целевой категории.

20.Cоздать столбец “Категория суммы кредитаˮ - Amount  (с помощью категоризации). Выделить минимум 3 категории (маленькая, крупная, средняя), фильтрацию для уровня суммы выбрать самостоятельно, аргументировать выбор. Создать столбец “Категория дохода (Income)ˮ  (с помощью категоризации). Выделить минимум 3 категории (низкий, высокий, средний), фильтрацию для уровня дохода выбрать самостоятельно, аргументировать выбор. Создать две группировки: медианная Emp_length по категории суммы кредита и медианная Emp_length  по категории дохода.

In [ ]:
emp_length_by_amount = bank.groupby('Amount_Category')['Employment_length'].median()

emp_length_by_income = bank.groupby('Income_Category')['Employment_length'].median()

print("Медианный стаж работы по категории суммы кредита:")
print(emp_length_by_amount)

print("\nМедианный стаж работы по категории дохода:")
print(emp_length_by_income)

Медианный стаж работы по категории суммы кредита:
Amount_Category
Крупная      4.0
Маленькая    3.0
Средняя      5.0
Name: Employment_length, dtype: float64

Медианный стаж работы по категории дохода:
Income_Category
Высокий    5.0
Низкий     3.0
Средний    4.0
Name: Employment_length, dtype: float64


*Важно заметить, что столбцы из задания создавались ранее в заданиях 3 и 9, поэтому этот этап пропускается, создаются только группировки*

Вывод

Проведенное исследование выявило устойчивую корреляцию между профессиональным стажем клиентов и их финансовыми характеристиками. Данные демонстрируют, что клиенты с медианным стажем 5 лет и более склонны к оформлению крупных кредитов и относятся к категории высокодоходных заемщиков. Напротив, клиенты с меньшим стажем работы (4 года) ограничиваются скромными кредитными суммами и демонстрируют более низкие показатели доходности.

Особенно показательным является факт идентичности медианного стажа для категорий "крупных кредитов" и "высоких доходов" - оба показателя составляют 5 лет. Это свидетельствует о том, что профессиональная устойчивость служит фундаментальным фактором, определяющим как уровень заработка, так и кредитную емкость клиентов.

Полученные результаты имеют стратегическое значение для банка, позволяя выстраивать более точные скоринговые модели и разрабатывать таргетированные продукты, учитывающие профессиональную стабильность заемщиков как ключевой критерий кредитоспособности.